_posts/2017-11-26-rust-vscode.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: post
+asset-type: post
+name: setting-up-vscode-for-rust
+title: Setting up Rust development environment using VSCode on a Mac
+description: A guide to setting up VSCode as the central point of your Rust development experience
+date: 2017-11-26 00:00:00 +00:00
+author: Cyryl Płotnicki
+image:
+   src: 
+tags:
+- rust
+- vscode
+---
+
+This post is a part of the upcoming series on different ways of setting up your Rust development environment. It's time for VSCode.
+
+While on Linux VSCode with the Rust plugin seems to work more or less out of the box, on a Mac I needed to spend some time configuring it.
+
+First things first though, let's start by installing Rust version manager, rustup.
+
+```
+curl https://sh.rustup.rs -sSf | sh
+```
+
+We will be using nightly version of rust as to have one version that can compile all of our tools. This is mostly due to [`clippy`](https://github.com/rust-lang-nursery/rust-clippy) requiring a nightly compiler.
+
+```
+rustup install nightly
+rustup default nightly
+```
+
+We will need Rust Language Server to provide the code completion.
+
+```
+rustup component add rls-preview --toolchain nightly
+rustup component add rust-analysis --toolchain nightly
+rustup component add rust-src --toolchain nightly
+```
+
+For a more wholesome experience, please have some tools as well:
+
+```
+cargo install clippy rustfmt rustsym
+```
+
+Now finally, for the VSCode itself, press `cmd-p` and `ext install vscode-rust`. I'm using the new `Rust` extension as `Rusty Code` has been discontinued.
+
+If you're lucky - that's it, you should have working completion and highlighting in Rust files.
```

Comment:
might be worth adding a line before telling them to open up a project in vs-code